scipy/optimize/_milp.py

```diff
@@ -0,0 +1,359 @@
+        Specified elements of x must be integers
+
+    By default, ``l = 0`` and ``u = np.inf`` unless specified with
+    ``bounds``.
+
+    Parameters
+    ----------
+    c : 1D array_like
+        The coefficients of the linear objective function to be minimized.
+        `c` is converted to a double precision array before the problem is
+        solved.
+    integrality : 1D array_like, optional
+        Indicates the type of integrality constraint on each decision variable.
+
+        ``0`` : Continuous variable; no integrality constraint.
+
+        ``1`` : Integer variable; decision variable must be an integer
+        within `bounds`.
+
+        ``2`` : Semi-continuous variable; decision variable must be within
+        `bounds` or take value ``0``.
+
+        ``3`` : Semi-integer variable; decision variable must be an integer
+        within `bounds` or take value ``0``.
+
+        By default, all variables are continuous. `integrality` is converted
+        to an array of integers before the problem is solved.
+    bounds : scipy.optimize.Bounds, optional
+        Bounds on the decision variables. Lower and upper bounds are converted
+        to double precision arrays before the problem is solved. The
+        ``keep_feasible`` parameter of the `Bounds` object is ignored. If
+        not specified, all decision variables are constrained to be
+        non-negative.
+    constraints : sequence of scipy.optimize.LinearConstraint, optional
+        Linear constraints of the optimization problem. Arguments may be
+        one of the following:
+
+        1. A single `LinearConstraint` object
+        2. A single tuple that can be converted to a `LinearConstraint` object
+           as ``LinearConstraint(*constraints)``
+        3. A sequence composed entirely of objects of type 1. and 2.
+
+        Before the problem is solved, all values are converted to double
+        precision, and the matrices of constraint coefficients are converted to
+        instances of `scipy.sparse.csc_array`. The ``keep_feasible`` parameter
+        of `LinearConstraint` objects is ignored.
+    options : dict, optional
+        A dictionary of solver options. The following keys
+        are recognized.
+
+        disp : bool (default: ``False``)
+            Set to ``True`` if indicators of optimization status are to be
+            printed to the console during optimization.
+        presolve : bool (default: ``True``)
+            Presolve attempts to identify trivial infeasibilities,
+            identify trivial unboundedness, and simplify the problem before
+            sending it to the main solver. It is generally recommended
+            to keep the default setting ``True``; set to ``False`` if
+            presolve is to be disabled.
+        time_limit : float
+            The maximum time in seconds allotted to solve the problem;
+            default is the largest possible value for a ``double`` on the
+            platform.
```

Comment:
I don't understand this statement. What is the largest possible value for a double in second?

Reply:
```suggestion
        time_limit : float
            The maximum number of seconds allotted to solve the problem;
            default is the largest possible value for a ``double`` on the
            platform.
```

Reply:
It's the second sentence that I don't get: `largest possible value for a double` I don't know what number this gets me on any platform.

Reply:
> It's the second sentence that I don't get: largest possible value for a double I don't know what number this gets me on any platform.

That's the point -- it's platform specific, i.e. in C++ it's `std::numeric_limits<int>::max()`.  This turns out to functionally be no timeout and I believe should be an astronomically large number

Reply:
Ah ok it does really mean this. I was not sure as it seemed indeed very large. Shouldn't we say there is no limit then? Or am I missing a use case where this distinction would matter? To me it then reads like anyone running this would be long dead before it finishes no matter the platform 😅

Reply:
On my platform, 
```
>>> np.finfo(float).max
1.7976931348623157e+308
```

If that is the number of seconds, then it will be about 5.7 x 10^300 years. According to [this](https://en.wikipedia.org/wiki/Graphical_timeline_from_Big_Bang_to_Heat_Death#:~:text=This%20is%20the%20timeline%20of,106%20years%2C%20if%20protons%20decay.), the heat death of the univere is a mere 10^100 years away. So I'm guessing it won't matter much on any platform.

This was copied from `linprog`, which I'm guessing was copied from HiGHS. Some people are very strict about documenting _precisely_ how a function behaves. I think this is taken _ad absurdum_ here.

Reply:
Yeah exactly, it's not only the value itself, but also the platform specific information which then reads like inf or inf + n. So inf 😅 Non blocking, I personally prefer to make a small "lie" here to clarify the doc.

Reply:
I'll do that.

Reply:
Fixed in next commit.